### DISCUSSION TITLE
Quarkus & Lambda & Hibernate SQL

### DISCUSSION THREAD

**Discussion by vladaman** [quarkusio/quarkus#28424] [Q&A]
We are using Quarkus with AWS Lambda and Hibernate and found out that every time the .jar is started it scans DB for schema changes. Is there any way to avoid this? Since purpose of Lamdba functions is to start as soon as possible to minimize cold start.

We tried to disable generation using property `quarkus.hibernate-orm.database.generation=none` but this still scans the DB for changes - it just doesn't apply them.

[a comment by an automated account is not shown]

**Comment by yrodiere** [comment]
Can you please post some of the SQL being run?

Also, which database are you using?

I think Hibernate ORM will have to run *some* SQL to gather *some* metadata, because it's basically implemented in the JDBC driver (see https://github.com/hibernate/hibernate-orm/blob/5.6/hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java#L66-L70).

We've thought about avoiding it, but that's a bit hard because we would essentially have to hardcode some database-specific values in Hibernate ORM, for every supported version of every supported database. Not impossible, but clearly there's significant work involved.

But in any case, it does not seem normal that this would fetch the whole schema. I would expect that to be done lazily, and when Hibernate ORM doesn't need it, not to be done at all.
